test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +109,43 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+#
+# Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+#
+# Added as an effort to improve error handling and avoid full anyhow backtrace
+# https://github.com/neondatabase/neon/pull/7949
+#
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    endpoint = env.endpoints.create_start("main")
+
+    tenant_id = env.initial_tenant
+    timeline_id = env.initial_timeline
+
+    conn = endpoint.connect()
+    cur = conn.cursor()
+    cur.execute("CREATE TABLE foo (x integer)")
+    tbl = []
+    cur.execute("INSERT INTO foo VALUES(0)")
+    # Get the timestamp at UTC
+    after_timestamp = query_scalar(cur, "SELECT clock_timestamp()").replace(tzinfo=None)
+    tbl.append([0, after_timestamp])
+    cur.execute("INSERT INTO foo VALUES (-1)")
+
+    with env.pageserver.http_client() as client:
```

Comment:
I think this is bad that we use http_client as a context manager; it does not implement those methods. I know this is copypasted from some other test case, but lets just make this a normal variable.

HTTP client by nature is stateless, I don't know what started this idea of using it as a context manager.